core/charm/strategies.go

```diff
@@ -49,6 +50,13 @@ func AlwaysChecksum(string) bool {
 	return true
 }
 
+// WithChecksum validates a checksum against another checksum.
```

Comment:
The function name and return type is a bit misleading. Perhaps calling the type `ChecksumCheckFn` and this function `MatchChecksum` would be a bit cleaner.